code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,210 @@
+		else if(distance <= boom_sizes[2])
+			rock.gets_drilled(null,TRUE,2)
+		else if(distance <= boom_sizes[3])
+			rock.gets_drilled(null,TRUE,1)
+	for(var/mob/living/carbon/C in circlerange(location,boom_sizes[3]))
+		if(ishuman(C)) //working on everyone
+			var/distance = get_dist_euclidian(location,C)
+			C.flash_eyes()
+			C.Weaken((boom_sizes[2] - distance) * 1 SECONDS) //1 second for how close you are to center if you're in range
+			C.AdjustDeaf((boom_sizes[3] - distance) * 10 SECONDS)
+			var/obj/item/organ/internal/ears/ears = C.get_int_organ(/obj/item/organ/internal/ears)
+			if(istype(ears))
+				ears.receive_damage((boom_sizes[3] - distance) * 2) //something like that i guess. Mega charge makes 12 damage to ears if nearby
+			to_chat(C, "<span class='warning'><font size='2'><b>You are knocked down by the power of the mining charge!</font></b></span>")
+	qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/proc/explode() //c4 code
+	var/turf/location
+	if(target)
+		if(!QDELETED(target))
+			if(istype(target, /turf/))
+				location = get_turf(target)
+			else
+				location = get_atom_on_turf(target)
+			target.overlays -= image_overlay
+	else
+		location = get_atom_on_turf(src)
+	if(location)
+		explosion(location, boom_sizes[1], boom_sizes[2], boom_sizes[3], cause = src)
+		location.ex_act(2, target)
+	if(istype(target, /mob))
+		var/mob/M = target
+		M.gib()
+	qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/proc/override_safety()
+	hacked = TRUE
+	notify_admins = TRUE
+	boom_sizes[1] = round(boom_sizes[1]/3)	//lesser - 0, normal - 0, mega - 1; c4 - 0
+	boom_sizes[2] = round(boom_sizes[2]/3)	//lesser - 0, normal - 1, mega - 2; c4 - 0
+	boom_sizes[3] = round(boom_sizes[3]/1.5)//lesser - 2, normal - 3, mega - 5; c4 - 3
+
+/obj/item/grenade/plastic/miningcharge/deconstruct(disassembled = TRUE) //no gibbing a miner with pda bombs
+	if(!QDELETED(src))
+		qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/lesser
+	name = "mining charge"
+	desc = "A mining charge. This one seems less powerful than industrial. Only works on rocks!"
+	icon_state = "mining-charge-1"
+	smoke_amount = 1
+	boom_sizes = list(1,2,3)
+
+/obj/item/grenade/plastic/miningcharge/mega
+	name = "experimental mining charge"
+	desc = "A mining charge. This one seems much more powerful than normal!"
+	icon_state = "mining-charge-3"
+	smoke_amount = 5
+	boom_sizes = list(4,6,8)
+
+/obj/item/storage/backpack/duffel/miningcharges/populate_contents()
+	for(var/i in 1 to 5)
+		new /obj/item/grenade/plastic/miningcharge/lesser(src)
+	for(var/i in 1 to 3)
+		new /obj/item/grenade/plastic/miningcharge(src)
+	new /obj/item/detonator(src)
+
+//MINING CHARGE HACKER
+
+/obj/item/t_scanner/adv_mining_scanner/syndicate
+	var/charges = 6
+	desc = "A scanner that automatically checks surrounding rock for useful minerals; it can also be used to stop gibtonite detonations. Wear meson scanners for optimal results. This scanner has an extra port for overriding mining charge safeties."
```

Comment:
```suggestion
	desc = "A scanner that automatically checks surrounding rock for useful minerals; it can also be used to stop gibtonite detonations. Wear meson scanners for optimal results. This scanner has an extra port for overriding mining charge safeties."
	
	var/charges = 6
```
inherited vars are defined first